app/src/main/java/openfoodfacts/github/scrachx/openfood/views/category/activity/CategoryActivity.java

```diff
@@ -16,10 +18,28 @@ public static Intent getIntent(Context context) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_category);
+        checkNetworkAndShowScreen();
+    }
+
+    private void checkNetworkAndShowScreen(){
+        ConnectivityManager cm =
+                (ConnectivityManager)this.getSystemService(Context.CONNECTIVITY_SERVICE);
+
+        NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
+        boolean isConnected = activeNetwork != null &&
```

Comment:
There is a  Helper Method in Utils.utils   named as `isNetworkConnected(Context context)`   You could use that and reduce this further.